lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1621,6 +1626,10 @@ private function validateAndCompleteTypedFieldMapping(array $mapping): array
                     case 'string':
                         $mapping['type'] = Types::STRING;
                         break;
+                    default:
+                        if (isset($this->typedFieldMappings[$type->getName()])) {
```

Comment:
I think `getName()` is deprecated. I don't think we should use it in a new feature, right?

Reply:
this is `ReflectionNamedType` https://www.php.net/manual/en/reflectionnamedtype.getname.php

I think you confused this with `Doctrine\DBAL\Types\Type`

Reply:
Ah yes indeed, sorry for the misunderstanding!